main/src/cgeo/geocaching/models/Waypoint.java

```diff
@@ -518,4 +518,14 @@ public int getMapDotMarkerBackgroundId() {
         return ConnectorFactory.getConnector(geocode).getCacheMapDotMarkerBackgroundId();
     }
 
+    public boolean applyDistanceRule() {
+        boolean applyDistanceRule = getWaypointType().applyDistanceRule();
+        if (applyDistanceRule) {
+            final Geocache parentCache = getParentGeocache();
+            if (parentCache != null) {
+                applyDistanceRule = parentCache.applyDistanceRule();
+            }
```

Comment:
I guess this will not work in the way it should work. `Geocache.applyDistanceRule()` will only return true if the cache type is Tradi or the cache has corrected coordinates.

However, we explicitly want to show the circle for final waypoints which do not belong to a tradi. Maybe change your code to the following?:

```suggestion
            if (parentCache != null && parentCache.isArchived()) {
                applyDistanceRule = false;
            }
```

Reply:
How about the connector-stuff?

Reply:
> How about the connector-stuff?

Hm, that might be worth to keep. However, it would be bad if these decisions need to magically stay in sync between different classes. So maybe create a method `Geocache.mayApplyDistanceRule()` which can be used inside `Geocache.applyDistanceRule()` as well as `Waypoint.applyDistanceRule()`?